packages/react-reconciler/src/__tests__/ReactLazy-test.internal.js

```diff
@@ -1329,11 +1329,7 @@ describe('ReactLazy', () => {
     expect(ref.current).toBe(null);
 
     await act(() => resolveFakeImport(Foo));
-    assertLog([
-      'Foo',
-      // pre-warming
-      'Foo',
-    ]);
+    assertLog(['Foo']);
```

Comment:
Now a Lazy gets to participate in the optimization when it can be immediately resolved in place instead of causing a rerender.

This doesn't have any of the other legacy quirks like infinite pinging concerns since these are unconditionally rendered in the child position and never inside a component render.